src/storage/src/hummock/state_store.rs

```diff
@@ -293,6 +305,32 @@ impl HummockStorage {
                 }
             }
         }
+        for sync_uncommitted_data in sync_uncommitted_datas.into_iter().rev() {
+            for data_list in sync_uncommitted_data {
+                for data in data_list {
+                    match data {
+                        UncommittedData::Batch(batch) => {
+                            if let Some(v) = get_from_batch(&batch) {
+                                return Ok(v);
+                            }
+                        }
+                        UncommittedData::Sst((_, table_info)) => {
+                            let table = self
+                                .sstable_store
+                                .sstable(table_info.id, &mut stats)
+                                .await?;
+                            table_counts += 1;
+                            if let Some(v) = self
+                                .get_from_table(table, &internal_key, key, &mut stats)
+                                .await?
+                            {
+                                return Ok(v);
+                            }
```

Comment:
These logics are the same as L285-302. Can we avoid duplicates codes by extracting them in a function?